BackEnd/Data/IUser.cs

```diff
@@ -5,18 +5,22 @@ namespace BackEnd.Data
 {
     public interface IUser
     {
-        User Authenticate(string username,string password);
+        User Authenticate(string username, string password);
         IEnumerable<User> GetAllUsers();
         User GetUserById(int id);
-        IEnumerable<Task> GetAllTasks(int id);
-        Task GetOneTask(int id, int taskId);
-        IEnumerable<Task> AddTask(int id, Task task);
-        Task EditTask(int id, int taskId, TaskModel task);
-        void DeleteTask(int id, int TaskId);
         User CreateUser(User user, string password);
         void Update(User user);
         void Delete(int id);
-        IEnumerable<TasksPerUsers> GetAll();
-
+        IEnumerable<object> TasksPerUsers(int id);
```

Comment:
You should separate this in multiple interfaces/repositories.
- 1 repository for all User related operations
- 1 repository for all Project related operations
- ...